src/main/java/com/redhat/devtools/intellij/tektoncd/actions/triggers/CreateTriggerTemplateAction.java

```diff
@@ -13,25 +13,55 @@
 import com.google.common.base.Strings;
 import com.intellij.openapi.actionSystem.AnActionEvent;
 import com.redhat.devtools.intellij.tektoncd.actions.TektonAction;
+import com.redhat.devtools.intellij.tektoncd.telemetry.TelemetryService;
 import com.redhat.devtools.intellij.tektoncd.tkn.Tkn;
 import com.redhat.devtools.intellij.tektoncd.tree.TriggerTemplatesNode;
 import com.redhat.devtools.intellij.tektoncd.utils.VirtualFileHelper;
+import com.redhat.devtools.intellij.telemetry.core.service.TelemetryMessageBuilder.ActionMessageBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import javax.swing.tree.TreePath;
 
+import java.io.IOException;
+
+import static com.redhat.devtools.intellij.tektoncd.Constants.KIND_CONDITIONS;
 import static com.redhat.devtools.intellij.tektoncd.Constants.KIND_TRIGGERTEMPLATES;
+import static com.redhat.devtools.intellij.tektoncd.telemetry.TelemetryService.PROP_RESOURCE_KIND;
+import static com.redhat.devtools.intellij.telemetry.core.util.AnonymizeUtils.anonymizeResource;
 
 public class CreateTriggerTemplateAction extends TektonAction {
-    public CreateTriggerTemplateAction() { super(TriggerTemplatesNode.class); }
+
+    private static final Logger logger = LoggerFactory.getLogger(CreateTriggerTemplateAction.class);
+
+    public CreateTriggerTemplateAction() {
+        super(TriggerTemplatesNode.class);
+    }
 
     @Override
     public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, Tkn tkncli) {
+        ActionMessageBuilder telemetry = TelemetryService.instance()
+                .action("create trigger template")
+                .property(PROP_RESOURCE_KIND, KIND_TRIGGERTEMPLATES);
         TriggerTemplatesNode item = getElement(selected);
         String namespace = item.getParent().getName();
         String content = getSnippet("Tekton: TriggerTemplate");
 
-        if (!Strings.isNullOrEmpty(content)) {
-            VirtualFileHelper.createAndOpenVirtualFile(anActionEvent.getProject(), namespace, namespace + "-newtriggertemplate.yaml", content, KIND_TRIGGERTEMPLATES, item);
+        if (Strings.isNullOrEmpty(content)) {
+            telemetry
+                    .error("snippet content empty")
+                    .send();
+        } else {
+            String name = namespace + "-newtriggertemplate.yaml";
+            try {
+                VirtualFileHelper.createAndOpenVirtualFile(anActionEvent.getProject(), namespace, name, content, KIND_TRIGGERTEMPLATES, item);
+                telemetry.send();
```

Comment:
`telemetry.success().send()`

Reply:
#success not required any more. One might argue on being more explicit though.